AdvanceDS/Assignment 1/genralize list/GenralizeList.java

```diff
@@ -0,0 +1,228 @@
+package addsAssignment;
+
+import java.util.List;
+import java.util.ArrayList;
+
+public class GenralizeList {
+	private Node start;
+	private Stack expression;
+	private int max = 0;
+	private int sum = 0;
+	private String outputList = "";
+	private boolean b = false;
+
+	public Node getStart() {
+		return start;
+
+	}
+
+	class Stack {
+		private int top;
+		private int size;
+		private Object stack[];
+
+		public Stack(int size) {
+
+			this.size = size;
+			top = -1;
+			stack = new Object[size];
+		}
+
+		public int getTop() {
+			return top;
+		}
+
+		public void setTop(int top) {
+			this.top = top;
+		}
+
+		public void push(Object item) {
+
+			if (top >= size - 1) {
+				System.out.println("Stack overflow");
+
+			} else {
+				top = top + 1;
+				stack[top] = item;
+			}
+
+		}
+
+		public Object pop() {
+			Object elementToPop = null;
+			if (size == -1) {
+				System.out.println("Stack underflow");
+			} else {
+
+				elementToPop = stack[top];
+				top--;
+
+			}
+
+			return elementToPop;
+		}
+
+		public boolean isEmpty() {
+			if (top == -1)
+				return true;
+			else
+				return false;
+
+		}
+
+		public Object peep() {
+			Object t = null;
+			if (top >= size) {
+				System.out.println("stack overflow");
+			} else if (top == -1) {
+
+				System.out.println("Stack underflow");
+			} else {
+				t = stack[top];
+			}
+			return t;
+		}
+
+	}
+
+	class Node {
+		Node next;
+		Object value;
+
+		public Node(Object item) {
+			value = item;
+			next = null;
+		}
+
+		public Node() {
+		}
+	}
+
+	public GenralizeList() {
+		start = null;
+	}
+
+	public GenralizeList(String list) {
+		expression = new Stack(list.length());
```

Comment:
I think you could have done this using recursion with less no. of lines and logic more clear.